src/python_testing/TC_CNET_4_10.py

```diff
@@ -0,0 +1,260 @@
+        else:
+            expected_network_id = thread_network_id_bytes
+            asserts.assert_equal(last_network_id, expected_network_id, "LastNetworkID incorrect when Networks list is not empty")
+
+        # Step 7: Verify breadcrumb
+        self.step(7)
+        breadcrumb = await self.read_single_attribute_check_success(
+            cluster=gen_comm,
+            attribute=gen_comm.Attributes.Breadcrumb
+        )
+        asserts.assert_equal(breadcrumb, 1)
+
+        # Step 8: Try to connect to removed network
+        self.step(8)
+        response = await self.send_single_cmd(
+            cmd=cnet.Commands.ConnectNetwork(
+                networkID=thread_network_id_bytes,
+                breadcrumb=2
+            )
+        )
+        asserts.assert_equal(response.networkingStatus,
+                             cnet.Enums.NetworkCommissioningStatusEnum.kNetworkIDNotFound)
+
+        # Step 9: Verify breadcrumb unchanged
+        self.step(9)
+        breadcrumb = await self.read_single_attribute_check_success(
+            cluster=gen_comm,
+            attribute=gen_comm.Attributes.Breadcrumb
+        )
+        asserts.assert_equal(breadcrumb, 1)
+
+        # Step 10: Disable failsafe
+        self.step(10)
+        await self.send_single_cmd(
+            cmd=gen_comm.Commands.ArmFailSafe(expiryLengthSeconds=0, breadcrumb=0)
+        )
+        # Successful command execution is implied if no exception is raised.
+
+        # Step 11: Verify network restored
+        self.step(11)
+        networks = await self.read_single_attribute_check_success(
+            cluster=cnet,
+            attribute=cnet.Attributes.Networks
+        )
+        asserts.assert_equal(len(networks), num_networks,
+                             "Network count not restored")
+        found = False
+        for network in networks:
+            if (network.networkID == thread_network_id_bytes and
+                    network.connected):
+                found = True
+                break
+        asserts.assert_true(found, "Thread network not restored")
+
+        # Step 12: Re-arm failsafe
+        self.step(12)
+        await self.send_single_cmd(
+            cmd=gen_comm.Commands.ArmFailSafe(expiryLengthSeconds=900, breadcrumb=0)
+        )
+        # Successful command execution is implied if no exception is raised.
+
+        # Step 13: Remove network again
+        self.step(13)
+        response = await self.send_single_cmd(
+            cmd=cnet.Commands.RemoveNetwork(
+                networkID=thread_network_id_bytes,
+                breadcrumb=1
+            )
+        )
+        asserts.assert_equal(response.networkingStatus,
+                             cnet.Enums.NetworkCommissioningStatusEnum.kSuccess)
+        asserts.assert_equal(response.networkIndex, userth_netidx)
+
+        # Step 14: Complete commissioning
+        self.step(14)
```

Comment:
I know this is part of the test plan but ... what? How does this work? This is the network you commissioned the device onto, right? The device is USING this network to communicate with the TH. You just removed it. So how are you still communicating? Does the spec not say the network should be disconnected when it is removed?

Reply:
Oh, actually, no, this makes sense otherwise you could never replace networks. OK, disregard.